homeassistant/components/hassio/backup.py

```diff
@@ -369,8 +371,9 @@ def on_job_progress(data: Mapping[str, Any]) -> None:
             await backup_complete.wait()
         finally:
             unsub()
-        if not backup_id:
-            raise BackupReaderWriterError("Backup failed")
+        if not backup_id or create_errors:
+            # We should add more specific error handling here in the future
+            raise BackupReaderWriterError(f"Backup failed: {create_errors}")
```

Comment:
If there's no `backup_id` I'm assuming `create_errors` could be empty. Should we do something about that?